src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -768,15 +768,38 @@ void Graph::AllocateWithReuse(const std::vector<size_t>& syncNodesInds) {
     // Markup the boxes
     std::vector<ov::MemorySolver::Box> definedBoxes;
     std::vector<ov::MemorySolver::Box> undefinedBoxes;
+
+    //define the memory allocation strategy
+    std::function<bool(const MemoryDesc&)> sizeDefined;
```

Comment:
Maybe just use pointers to member functions instead of `std::function`?

```cpp
    const auto sizeDefined = flag ? &MemoryDesc::hasDefinedMaxSize : &MemoryDesc::isDefined;
    const auto calcSize = flag ? &MemoryDesc::getMaxMemSize : &MemoryDesc::getCurrentMemSize;
    
    // And use it
    (desc.*sizeDefined)()
```

Reply:
Good suggestion, thank you!